flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -165,6 +168,41 @@ public Options getCustomCommandLineOptions() {
 	//  Execute Actions
 	// --------------------------------------------------------------------------------------------
 
+	protected void runApplication(String[] args) throws Exception {
+		LOG.info("Running 'run-application' command.");
+
+		final Options commandOptions = CliFrontendParser.getRunCommandOptions();
+		final CommandLine commandLine = getCommandLine(commandOptions, args, true);
+
+		final ProgramOptions programOptions = new ProgramOptions(commandLine);
+
+		if (commandLine.hasOption(HELP_OPTION.getOpt())) {
+			CliFrontendParser.printHelpForRun(customCommandLines);
+			return;
+		}
+
+		final ApplicationDeployer deployer =
+				new ApplicationClusterDeployer(clusterClientServiceLoader);
+
+		final PackagedProgram program =
+				getPackagedProgram(programOptions);
+
+		final ApplicationConfiguration applicationConfiguration =
+				new ApplicationConfiguration(program.getArguments(), program.getMainClassName());
+
+		try {
+			final List<URL> jobJars = program.getJobJarAndDependencies();
```

Comment:
I agree, and this is something that I also have in my mind. But I think that this can be done in the follow-up issue of being able to specify (remote) paths and the packaged program is going to be created on the JM. What do you think?

Reply:
Yes, it could be done in the following tickets.